packages/@dcl/ecs/src/systems/crdt/index.ts

```diff
@@ -164,32 +164,53 @@ export function crdtSceneSystem(engine: Pick<IEngine, 'getComponentOrNull'>) {
       }
     }
 
-    // Send messages to transports
+    // Send CRDT messages to transports
     const transportBuffer = createByteBuffer()
     for (const transport of transports) {
       transportBuffer.resetBuffer()
+      // First we need to send all the messages that were outdated from a transport
+      // So we can fix their crdt state
+      for (const message of outdatedMessages) {
+        if (message.transportType === transport.type) {
+          transportBuffer.writeBuffer(message.messageBuffer, false)
+        }
+      }
+      // Then we send all the new crdtMessages that the transport needs to process
       for (const message of crdtMessages) {
         if (transport.filter(message)) {
           transportBuffer.writeBuffer(message.messageBuffer, false)
         }
       }
-      if (transportBuffer.size()) {
-        transport.send(transportBuffer.toBinary())
-      } else {
-        transport.send(new Uint8Array([]))
-      }
+      const message = transportBuffer.size()
+        ? transportBuffer.toBinary()
+        : new Uint8Array([])
+      await transport.send(message)
     }
   }
 
+  /**
+   * @public
+   * Add a transport to the crdt system
+   */
   function addTransport(transport: Transport) {
+    if (transports.find((t) => transport.type === t.type)) {
+      throw new Error('Transport already added')
+    }
```

Comment:
What is the rationale behind this if? It seems like the addTransport should not care about anything related to the type of the transport. i.e. we sould be able to add 15 ws transports if desired